src/compass/s1_geocode_metadata.py

```diff
@@ -122,6 +126,131 @@ def run(cfg, burst, fetch_from_scratch=False):
+        Decimation factor to downsample the slant range pixels for LUT
+    '''
+    dem_raster = isce3.io.Raster(cfg.dem)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+    burst_id = str(burst.burst_id)
+    geo_grid = cfg.geogrids[burst_id]
+    radar_grid = burst.as_isce3_radargrid()
+
+    date_str = burst.sensing_start.strftime("%Y%m%d")
+    burst_id_date_key = (burst_id, date_str)
+    out_paths = cfg.output_paths[burst_id_date_key]
+
+    # Common initializations
+    threshold = cfg.geo2rdr_params.threshold
+    iters = cfg.geo2rdr_params.numiter
+    scratch_path = out_paths.scratch_directory
+
+    root_path = '/science/SENTINEL1'
+
+    # Designate radiometric calibration parameter to geocode
+    calibration_dict = {
+        'gamma':burst.burst_calibration.gamma,
+        'sigma_naught':burst.burst_calibration.sigma_naught,
+    }
+
+    # define the geogrid for calbration LUT
+    radargrid_calibration = radar_grid.multilook(dec_factor,
+                                                 dec_factor)
+    geogrid_calibration = isce3.product.GeoGridParameters(
+                            geo_grid.start_x,
+                            geo_grid.start_y,
+                            geo_grid.spacing_x * dec_factor,
+                            geo_grid.spacing_y * dec_factor,
+                            geo_grid.width // dec_factor + 1,
+                            geo_grid.length // dec_factor + 1,
+                            geo_grid.epsg)
+
+    geo_calibration = isce3.geocode.GeocodeFloat32()
+    geo_calibration.orbit = burst.orbit
+    geo_calibration.ellipsoid = ellipsoid
+    geo_calibration.doppler = isce3.core.LUT2d()
+    geo_calibration.threshold_geo2rdr = threshold
+    geo_calibration.numiter_geo2rdr = iters
+    geo_calibration.geogrid(geogrid_calibration.start_x,
+                            geogrid_calibration.start_y,
+                            geogrid_calibration.spacing_x,
+                            geogrid_calibration.spacing_y,
+                            geogrid_calibration.width,
+                            geogrid_calibration.length,
+                            geogrid_calibration.epsg)
+    dem_raster = isce3.io.Raster(cfg.dem)
+    calibration_group_path =\
+        f'{root_path}/CSLC/metadata/calibration_information'
+    calibration_group =\
+        geo_burst_h5.require_group(calibration_group_path)
+
+    drv_lut_radargrid = gdal.GetDriverByName('ENVI')
+    for calibration_key, vec_calib in calibration_dict.items():
+        init_geocoded_dataset(calibration_group,
+                                calibration_key,
+                                geogrid_calibration,
+                                'float32',
+                                f'geocoded {calibration_key}')
+
+        calibration_dataset =\
+            geo_burst_h5[f'{calibration_group_path}/{calibration_key}']
+
+        calibration_burst_raster =\
```

Comment:
```suggestion
        # prepare output raster
        geocoded_cal_lut_raster =\
```
comment and rename for clarity